File: pdu/javascript/hako_srv_msgs/pdu_cdr_conv_NotifyAssetTimeResponsePacket.js
import { NotifyAssetTimeResponsePacket } from './pdu_jstype_NotifyAssetTimeResponsePacket.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { NotifyAssetTimeResponse } from '../hako_srv_msgs/pdu_jstype_NotifyAssetTimeResponse.js';
import { PduNotifyAssetTimeResponseConverter } from '../hako_srv_msgs/pdu_cdr_conv_NotifyAssetTimeResponse.js';
import { ServiceResponseHeader } from '../hako_srv_msgs/pdu_jstype_ServiceResponseHeader.js';
import { PduServiceResponseHeaderConverter } from '../hako_srv_msgs/pdu_cdr_conv_ServiceResponseHeader.js';


export class PduNotifyAssetTimeResponsePacketConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { NotifyAssetTimeResponsePacket } src
     */
    static to_cdr_body(writer, src) {
        PduServiceResponseHeaderConverter.to_cdr_body(writer, src.header);
        PduNotifyAssetTimeResponseConverter.to_cdr_body(writer, src.body);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { NotifyAssetTimeResponsePacket } dst
     * @returns { NotifyAssetTimeResponsePacket }
     */
    static cdr_body_to_js(reader, dst) {
        PduServiceResponseHeaderConverter.cdr_body_to_js(reader, dst.header);
        PduNotifyAssetTimeResponseConverter.cdr_body_to_js(reader, dst.body);
        return dst;
    }

    /**
     * @param { NotifyAssetTimeResponsePacket } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { NotifyAssetTimeResponsePacket }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new NotifyAssetTimeResponsePacket());
    }
}

File: pdu/python/hako_srv_msgs/pdu_cdr_conv_NotifyAssetTimeResponsePacket.py
from .pdu_pytype_NotifyAssetTimeResponsePacket import NotifyAssetTimeResponsePacket
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..hako_srv_msgs.pdu_cdr_conv_NotifyAssetTimeResponse import *
from ..hako_srv_msgs.pdu_cdr_conv_ServiceResponseHeader import *



def py_to_cdr_body_NotifyAssetTimeResponsePacket(writer: CdrWriter, src: NotifyAssetTimeResponsePacket):
    py_to_cdr_body_ServiceResponseHeader(writer, src.header)
    py_to_cdr_body_NotifyAssetTimeResponse(writer, src.body)


def cdr_body_to_py_NotifyAssetTimeResponsePacket(reader: CdrReader, dst: NotifyAssetTimeResponsePacket):
    cdr_body_to_py_ServiceResponseHeader(reader, dst.header)
    cdr_body_to_py_NotifyAssetTimeResponse(reader, dst.body)
    return dst


def py_to_cdr_NotifyAssetTimeResponsePacket(src: NotifyAssetTimeResponsePacket) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_NotifyAssetTimeResponsePacket(writer, src)
    return writer.bytes()


def cdr_to_py_NotifyAssetTimeResponsePacket(cdr_payload) -> NotifyAssetTimeResponsePacket:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = NotifyAssetTimeResponsePacket()
    return cdr_body_to_py_NotifyAssetTimeResponsePacket(reader, dst)
